server/cluster/cluster.go

```diff
@@ -450,6 +451,12 @@ func (c *RaftCluster) runStoreConfigSync() {
 //   - `switchRaftV2` is true if the config of tikv engine is change to raft-kv2.
 func (c *RaftCluster) syncStoreConfig(stores []*core.StoreInfo) (synced bool, switchRaftV2 bool) {
 	for index := 0; index < len(stores); index++ {
+		select {
+		case <-c.ctx.Done():
+			log.Info("sync store config job is stopped")
```

Comment:
What about being different with line 442?

Reply:
it may wait too long if there are many unhealthy store.

Reply:
fixed